### ISSUE TITLE
get key/token failed from google

### ISSUE THREAD

**Issue by ChiseLai** [moonstar-x/discord-tts-bot#24]
Hello there, I have been using this bot for quite a long while and it was running smoothly for me and my friend who is unable to speak.
But recently the bot is having the problem of get key failed from google which I try to look up for solution but all I can find is tell me to npm update
I tried it a few times but it also happen quite often, so I would like to ask if there is any method to fix this issue other than npm update.

Thanks
![image](https://user-images.githubusercontent.com/63945206/97434365-dfa75000-1959-11eb-964a-6196fa1634e7.png)

**Comment by moonstar-x** [comment]
Hey there, I see your console has a CJK language font, are you located in China by any chance? Is your region blocking https://translate.google.com?

**Comment by ChiseLai** [comment]
> Hey there, I see your console has a CJK language font, are you located in China by any chance? Is your region blocking https://translate.google.com?

I’m actually living in Hong Kong so I don’t think it has blocked translate.google.com
Where the bot can still speak texts it will sometime stuck in this little bug and it needs someone to input another speak command again before it starts talking again

**Comment by moonstar-x** [comment]
Unfortunately, you're hitting this error:  https://github.com/zlargon/google-tts/issues/27 (Which is a bit out of my hands).

The issue hasn't been very well discussed either so there's no way to tell what's happening other than your ISP may be blocking some requests to https://translate.google.com. I know that Google isn't blocked in Hong Kong so it is a bit weird that you're getting that particularly error.

I'm gonna have to close this issue because it isn't technically related to this bot but with the `google-tts`  dependency which I cannot change because it's the only one that does Google TTS with no Google Cloud Platform account (which isn't free).

A solution I can offer you is to [add the bot  that I'm hosting](https://discordapp.com/oauth2/authorize?client_id=519207945318170654&scope=bot&permissions=3148800) to your own server. (Though you may need to wait a bit because I'm in the process of getting the bot verified)

**Comment by solyarisoftware** [comment]
BTW, I guess the issue is not related exclusively from the region: it's for "google-inside". See: https://github.com/zlargon/google-tts/issues/33 (I'm from Italy).

**Comment by moonstar-x** [comment]
You are quite right, apparently location isn't a factor here. My service is hosted in Ecuador and now that I decide to check its logs I spot this same error.

![image](https://user-images.githubusercontent.com/14969195/98134301-9c366f80-1e8c-11eb-974d-6f2428240c05.png)

The randomness of this issue is quite unfortunate, makes it more complicated to fix it. I'll see what I can do. (Thanks for the heads up @solyarisoftware )

**Comment by solyarisoftware** [comment]
BTW, I'm experiencing the error just now (last few weeks). 
Workaround (not rocket-science): because the google API rais error occasionally, 
in my case I retry few times until I got success. Here the ugly pseudocode:

```
for (;;) {
callGoogleAPI(...)
if success 
  break
sleep(few msecs)
}  
```

**Comment by moonstar-x** [comment]
Thanks for the suggestion, I'll implement this until a fix can be found on the original library.

**Comment by ChiseLai** [comment]
Oh my god thank you so much
I have suffered quite a lot by this bug for a week now, trying to debug in every single way with my limited knowledge of coding and API understandings
I have tried to change host from translate.google.com to translate.google.cn (Since I once believed that I might have to use the China one to fix it)
I also have tried to see if it is my network connection problems, tried to even get a new adapter and change IP to see if that is helpful
Also tried to turn on VPN to see if this might do something
Thank you so much for looking into this

**Comment by moonstar-x** [comment]
Hey there, sorry for the long time it took me to respond to you, school has been very time consuming and didn't even see your edited message.

I updated the dependencies in #26 because the creator of the google-tts library made an update to do the retry solution. All you need to do is pull the changes from master with 

```
git pull origin master
```

Then remove the node_modules folder and reinstall the dependencies.

```
rm -rf node_modules && npm install
```

Sorry once again for taking way too long with this.

**Comment by ChiseLai** [comment]
Sorry for mentioning this issue in this closed issue thread
but seems like the similar issue with the google-api module is happening again
and this time its related to the get **token** key failed from google issue
I do know it is better to open a new issue thread for this, but since I think its the same category issue so I have decided to put it here
Also, I found that this issue makes the bot unable to speak a single sentence, unlike the inconsistent error appearing last time
![image](https://user-images.githubusercontent.com/63945206/100703311-3af7b280-33de-11eb-9693-a6f8174dff0b.png)

**Comment by moonstar-x** [comment]
Thanks for the report.

It seems that Google is migrating their translate API and is rendering multiple libraries that made use of it useless.

What's great though is that it seems that there's already an effort being put onto finding a solution for this, so all that remains is to wait.

I'm sorry for the trouble that all this may be causing you but sadly this is an issue affecting plenty of users of the translate API. :(

I'm gonna leave this issue open as reference.

**Comment by Receb-TR** [comment]
npm google-tts-api@0.0.5 has get key error and no updates .-.

**Comment by ChiseLai** [comment]
Don't worry we can wait for it
And again, its already great to hear that people are starting to look for solutions on it :D

**Comment by longphuc110** [comment]
I fixed it by updating the google-tts-api version again. Try again: npm i google-tts-api@0.0.6

**Comment by moonstar-x** [comment]
Thanks for the  suggestion. I just checked how the change went and quite unfortunately it is provisional. I'm gonna update the package.json to have this version but I'm considering using another/writing my own library for the API change,

I'm gonna leave this issue still open because I know for a fact that the 0.0.6 version will break again.

**Comment by moonstar-x** [comment]
This issue is fixed in #28. Please update your bot by pulling the changes in master.

Sorry for taking too long, uni has been very demanding as of lately! :(
